docs/v2_proposal.rst

```diff
@@ -0,0 +1,39 @@
+=======================
+OCX-Library V2 Proposal
+=======================
+
+The intention of this library is to act as a wrapper around the underlying client libraries.
+A user will import all of the "python-[service]client" libraries they would like to use, in addition to ours. 
+Our library relies on the existance of the other client libraries in the python environment.
+
+Code control flow:
+------------------
+* First call
+    - initialize x keystone clients
+    - initialize x service clients, one for each keystone client
+    - call the function with kwargs on the x service clients
+* Every subsequent call, until clients time out
+    - retrieve the service clients stored in the user's running software 
+    - call the function with kwargs on the x service clients
+* Once clients time out, we inform the user, and ask if they would like to re-initialize the clients
+
+Necessary Functions:
+--------------------
+* init_keystone_clients(auth_url_list, user_name, password)
+    - return keystone_client_list
+* init_service_clients(keystone_client_list, service)
+    - return service_client_list [list of nova clients, neutron clients, etc...]
```

Comment:
we should definitely discuss on what services that we want to include in the OCX library. Keystone, Nova, Glance, Swift, Cinder and Neutron is doable, and we haven't explored the source code of Heat yet. Also, the devstack of OpenStack doesn't have object storage service, but our harvard cluster does. So does it affect our implementation of OCX library with Cinder?